MapboxNavigation/RouteMapViewController.swift

```diff
@@ -282,6 +308,25 @@ class RouteMapViewController: UIViewController {
     func mapView(_ mapView: MGLMapView, viewFor annotation: MGLAnnotation) -> MGLAnnotationView? {
         return navigationMapView(mapView, viewFor: annotation)
     }
+    
+    var cameraCurrentForRouteProgress: MGLMapCamera {
+        
+        guard !isInOverviewMode else {
+            return genericTiltedCamera
+        }
+        
+        guard let _ = routeController.routeProgress.currentLegProgress.upComingStep else {
+            return genericTiltedCamera
+        }
+        
+        if let _ = routeController.routeProgress.currentLegProgress.currentStep.intersections?.first?.outletRoadClasses?.contains(.motorway) {
+            return motorwayCamera
+        } else if routeController.routeProgress.currentLegProgress.alertUserLevel == .high {
```

Comment:
It's also worth calling out that since we're grabbing the currentLegProgress more than once, it may be worth extracting that out into a local variable (above 322) for reuse. IMHO this would improve readability even more.

```swift
let legProgress = routecontroller.routeProgress.currentLegProgress

if let intersection = legProgress.currentStep.intersections?.first,
   let outletTypes = intersection.outletRoadClasses,
   outletTypes.contains(.motorway) {
    return motorwayCamera
} else if legProgress.alertUserLevel == .high {
    return overheadCamera
} else {
    return genericTiledCamera
}
```